libnmstate/route_rule.py

```diff
@@ -162,6 +202,7 @@ def gen_metadata(self, route_state, ifaces):
             for rule in rules:
                 family = Interface.IPV6 if rule.is_ipv6 else Interface.IPV4
                 route_rule_metadata[iface_name][family].append(rule.to_dict())
+                ifaces[iface_name].mark_as_changed()
```

Comment:
With `self._cur_rules`, you can check whether desired rule changed from current rule for this table id, and only mark as changed if so.

Meanwhile, if someone removed the all rule from certain table ID, that interface will not be marked as changed. But this is little bit complex and can wait in another PR.